### PR TITLE
Upgrade to julia 1.9

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR upgrades to julia 1.9

Blockers
- [x] Get latest TaylorSeries.jl: https://github.com/CliMA/CubedSphere.jl/pull/29
- [x] https://github.com/JuliaSIMD/HostCPUFeatures.jl/issues/12 (required to get `JULIA_CPU_TARGET` to work properly)

### CODE REVIEW

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by charleskawczynski** [comment]
I went ahead and canceled the job that never seemed to start after 7 hours. Looks like this is ready though, I’m looking forward to upgrading!

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
   slurm_mem: 8G
-  modules: julia/1.8.5 cuda/11.8 ucx/1.14.1_cuda-11.8 openmpi/4.1.5_cuda-11.8 hdf5/1.12.2-ompi415 nsight-systems/2022.2.1
+  modules: julia/1.9.0 cuda/11.8 ucx/1.14.1_cuda-11.8 openmpi/4.1.5_cuda-11.8 hdf5/1.12.2-ompi415 nsight-systems/2022.2.1
 
 env:
   JULIA_LOAD_PATH: "${JULIA_LOAD_PATH}:${BUILDKITE_BUILD_CHECKOUT_PATH}/.buildkite"
   OPENBLAS_NUM_THREADS: 1
   JULIA_NVTX_CALLBACKS: gc
   OMPI_MCA_opal_warn_on_missing_libcuda: 0
+  JULIA_CPU_TARGET: broadwell # arch used by GPU nodes
```
Comment (charleskawczynski) [inline .buildkite/pipeline.yml:11]: Where did you find that this env key values was needed?

File: examples/Project.toml [hunk 2]
```diff
@@ -32,10 +31,5 @@ Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 ThreadsX = "ac1d9e8a-700a-412c-b207-f0111f4b6c0d"
 UnPack = "3a884ed6-31ef-47d7-9d2a-63182c4928ed"
 
-[compat]
-OrdinaryDiffEq = "5.64.0"
```
Comment (charleskawczynski) [inline examples/Project.toml:36]: How is this suddenly not needed anymore? I thought that there were interface changes we needed to upgrade for the implicit solver. Are we not running any implicit examples? If so, that’s fine.

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (simonbyrne) [reply .buildkite/pipeline.yml:11]: The problem is that we may precompile on one node, and run on another, which invalidates a bunch of caches: `broadwell` are the oldest cores we have on the cluster.

See https://github.com/JuliaLang/julia/issues/50148 for some related discussion

File: examples/Project.toml (hunk 2 above)
Comment (simonbyrne) [reply examples/Project.toml:36]: Yeah, i was just trying to get the most recent versions. I'm not sure how best to deal with this (we may have to bite the bullet and update everything)

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try-

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
I just rebased and reverted the specific package adds (using PackageSpec), since that page seems to have been merged.

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
Whoops 😬, I forgot to look before force-pushing, I hope I didn't delete anything

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (simonbyrne) [reply .buildkite/pipeline.yml:11]: Could also change it to
```suggestion
  JULIA_CPU_TARGET: 'broadwell;skylake'
```

Comment (charleskawczynski) [reply .buildkite/pipeline.yml:11]: Thanks, I'll try this next after retrying more recent versions

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
🤞🏻

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (charleskawczynski) [reply .buildkite/pipeline.yml]: This also did not work, at least in that CI still failed. xref issue: https://github.com/JuliaCI/julia-buildkite/issues/298

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Pull request closed without merging**